x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -270,6 +270,10 @@ export function buildComponentTemplates(params: {
 
   const indexTemplateSettings = registryElasticsearch?.['index_template.settings'] ?? {};
 
+  // TODO: Is this the place that index_mode should be added as it belongs into template.settings.index.mode
+  // On package upgrade, what happens if the previous package version had index_mode: time_series and now it is
+  // switched back? This would work for the template but not when the data stream is rolled over
+  // Ideally TSDB can be disabled again but is not possible today -> validation on the package side needed?
```

Comment:
It's a good point that we could set `index.mode` as normal here, but the next rollover would fail. Not a very good experience around upgrading package versions in that case. We could probably catch the case where a package tried to disable `tsdb` after enabling it in a previous version here and throw an error, but it'd result in a package version that cannot be upgraded to. So, it makes the most sense to add some package spec validation here blocking a release that includes a change from `index_mode: time_series` -> default.

Reply:
I talked to the Elasticsearch team about the issue that tsdb currently can't be disabled. https://github.com/elastic/elasticsearch/issues/92544 was opened to make it possible to turn it off again. If this lands, this problem should go away.